lib/graphql/schema/field.rb

```diff
@@ -102,16 +102,16 @@ def initialize(type: nil, name: nil, owner: nil, null: nil, field: nil, function
         if name.nil?
           raise ArgumentError, "missing first `name` argument or keyword `name:`"
         end
-        if !(field || function || mutation || resolver)
+        if !(field || function || resolver_class)
           if type.nil?
             raise ArgumentError, "missing second `type` argument or keyword `type:`"
           end
           if null.nil?
             raise ArgumentError, "missing keyword argument null:"
           end
         end
-        if (field || function || resolve || mutation) && extras.any?
-          raise ArgumentError, "keyword `extras:` may only be used with method-based resolve, please remove `field:`, `function:`, `resolve:`, or `mutation:`"
+        if (field || function || resolve) && extras.any?
+          raise ArgumentError, "keyword `extras:` may only be used with method-based resolve and class-based field such as mutation class, please remove `field:`, `function:` or `resolve:`"
```

Comment:
The keyword argument `extras` has been supported too in class-based fields since #1313. So I've removed `mutation` from here and edited the error message.